File: sample/src/main/java/com/readdle/swiftjava/sample/SampleEnumWithValue.kt
package com.readdle.swiftjava.sample

import com.readdle.codegen.anotation.SwiftFunc
import com.readdle.codegen.anotation.SwiftValue

@SwiftValue
sealed class SampleEnumWithValue {

    object None : SampleEnumWithValue()
    data class StringValue(val value: String) : SampleEnumWithValue()
    data class IntegerValue(val value: Int) : SampleEnumWithValue()
    data class Int8Value(val value: Byte) : SampleEnumWithValue()
    data class Int16Value(val value: Short) : SampleEnumWithValue()
    data class Int32Value(val value: Int) : SampleEnumWithValue()
    data class Int64Value(val value: Long) : SampleEnumWithValue()
    data class UintValue(val value: UInt) : SampleEnumWithValue()
    data class Uint8Value(val value: UByte) : SampleEnumWithValue()
    data class Uint16Value(val value: UShort) : SampleEnumWithValue()
    data class Uint32Value(val value: UInt) : SampleEnumWithValue()
    data class Uint64Value(val value: ULong) : SampleEnumWithValue()
    data class ObjectArrayValue(val value: ArrayList<SampleValue>) : SampleEnumWithValue()
    data class StringArrayValue(val value: ArrayList<String>) : SampleEnumWithValue()
    data class NumberArrayValue(val value: ArrayList<Int>) : SampleEnumWithValue()
    data class ArrayInArrayValue(val value: ArrayList<ArrayList<Int>>) : SampleEnumWithValue()
    data class DictInArrayValue(val value: ArrayList<HashMap<Int, Int>>) : SampleEnumWithValue()
    data class DictSampleClassValue(val value: HashMap<String, SampleValue>) : SampleEnumWithValue()
    data class DictStringsValue(val value: HashMap<String, String>) : SampleEnumWithValue()
    data class DictNumbersValue(val value: HashMap<Int, Int>) : SampleEnumWithValue()
    data class Dict64NumbersValue(val value: HashMap<Int, Int>) : SampleEnumWithValue()
    data class DictInDictValue(val value: HashMap<Int, HashMap<Int, Int>>) : SampleEnumWithValue()
    data class ArrayInDictValue(val value: HashMap<Int, ArrayList<Int>>) : SampleEnumWithValue()
    data class SetValue(val value: HashSet<Int>) : SampleEnumWithValue()
    data class SetValuesValue(val value: HashSet<SampleValue>) : SampleEnumWithValue()

    companion object {

        @JvmStatic
        @SwiftFunc("copy(value:)")
        external fun copy(value: SampleEnumWithValue): SampleEnumWithValue
    }
}

File: sample/src/main/swift/Sources/SampleAppCore/SampleEnumWithValue.swift
//
// Created by Andrew Druk on 06.05.2024.
//

import Foundation

public enum SampleEnumWithValue: Codable {
    case none
    case stringValue(value: String)
    case integerValue(value: Int)
    case int8Value(value: Int8)
    case int16Value(value: Int16)
    case int32Value(value: Int32)
    case int64Value(value: Int64)
    case uintValue(value: UInt)
    case uint8Value(value: UInt8)
    case uint16Value(value: UInt16)
    case uint32Value(value: UInt32)
    case uint64Value(value: UInt64)
    case optionalIntegerValue(value: Int?)
    case optionalInt8Value(value: Int8?)
    case optionalInt16Value(value: Int16?)
    case optionalInt32Value(value: Int32?)
    case optionalInt64Value(value: Int64?)
    case optionalUintValue(value: UInt?)
    case optionalUint8Value(value: UInt8?)
    case optionalUint16Value(value: UInt16?)
    case optionalUint32Value(value: UInt32?)
    case optionalUint64Value(value: UInt64?)
    case objectArrayValue(value: [SampleValue])
    case stringArrayValue(value: [String])
    case numberArrayValue(value: [Int])
    case arrayInArrayValue(value: [[Int]])
    case dictInArrayValue(value: [[Int: Int]])
    case dictSampleClassValue(value: [String: SampleValue])
    case dictStringsValue(value: [String: String])
    case dictNumbersValue(value: [Int: Int])
    case dict64NumbersValue(value: [Int: Int])
    case dictInDictValue(value: [Int: [Int: Int]])
    case arrayInDictValue(value: [Int: [Int]])
    case setValue(value: Set<Int>)
    case setValuesValue(value: Set<SampleValue>)

    public static func copy(value: SampleEnumWithValue) -> SampleEnumWithValue {
        return value
    }
}